### PR TITLE
Added chat and chat list use case

### PR DESCRIPTION
**Pull request by aFluffyHotdog**
Completed the use cases: chat, list chat

### CODE REVIEW

File: src/main/java/app/ListChatUseCaseFactory.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package app;
```
Comment (puyepuye) [inline src/main/java/app/ListChatUseCaseFactory.java:1]: Great Clean Architecture Practices to make UseCases! Wow

File: src/main/java/use_case/navbar/NavbarInputBoundary.java
Comment (puyepuye) [inline src/main/java/use_case/navbar/NavbarInputBoundary.java:1]: Looking good, good catch on the debugged error.

File: src/main/java/data_access/SpringUserDAO.java
Comment (puyepuye) [inline src/main/java/data_access/SpringUserDAO.java:1]: This has been moved to main, I assume.

**Review by puyepuye — APPROVED** [review]
# Code Review: Chat and Chat List Use Case

Overall: Very good code in a time crunch!

## Coding Style
- **Strengths**: Meaningful variable names (`chatController`, `chatViewModel`) and consistent formatting.
- **Improvements**: 
  - `ChatView` class combines UI rendering and logic. Consider separating responsibilities.
  - Add comments for UI logic, such as hardcoded layout properties.

## Correctness
- **Strengths**: 
  - `addPropertyChangeListener` for `chatViewModel` is implemented well.
- **Concerns**: 
  - Ensure `chatURL` and `currentUser` are initialized properly.
  - Hardcoded layout properties (e.g., `setMaximumSize`) might not adapt well to different devices.

## Tests and Test Coverage
- Add unit tests for `ChatViewModel` to validate state changes and event triggering.
- Include integration tests for `ChatController` and navigation logic.

## Clean Architecture + SOLID
- **Strengths**: 
  - Clear separation between `ChatController`, `ChatView`, and `ChatViewModel`.
  
## Javadoc
- Sufficient documentation has been added. 

## Clarifications
- **Navigation Logic**: How does the `backButton` interact with other parts of the app?
- **Error Handling**: Are there safeguards for chat state conflicts or failed data loading?

**Comment by aFluffyHotdog** [comment]
1.`backButton` takes you back to the chatList page, with the navbar so that you can navigate to other pages
2. If data fails to load, the user can leave the page and enter again to try and reload

File: src/main/java/view/ListChatView.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+import javax.imageio.ImageIO;
+import javax.swing.*;
+import javax.swing.border.EmptyBorder;
+import java.awt.*;
+import java.awt.event.*;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.io.IOException;
+import java.net.URL;
+import java.util.ArrayList;
```
Comment (YolandaThant) [inline src/main/java/view/ListChatView.java:19]: JavaDoc: Missing JavaDoc. While it is understandable that the ListChatView class represent the GUI for displaying a list of chat channels by integrating with the ListChatViewModel, ListChatController, & NavbarController files, it is considered good practice to include a JavaDoc comment. This promotes clear communication and readability for future collaborators who may contribute to this project.

**Review by YolandaThant — COMMENTED** [review]
JavaDoc: Missing JavaDoc. While it is understandable that the ListChatView class represent the GUI for displaying a list of chat channels by integrating with the ListChatViewModel, ListChatController, & NavbarController files, it is considered good practice to include a JavaDoc comment. This promotes clear communication and readability for future collaborators who may contribute to this project.

File: src/main/java/app/NavbarUseCaseFactory.java [hunk 3]
```diff
@@ -44,10 +45,12 @@ public static NavBarView create(
             ViewManagerModel viewManagerModel,
             SwipeViewModel swipeViewModel,
             NavbarViewModel navBarViewModel,
-            CompatibilityViewModel compatibilityViewModel) {
+            CompatibilityViewModel compatibilityViewModel,
+            ListChatViewModel listChatViewModel) {
```
Comment (YolandaThant) [inline src/main/java/app/NavbarUseCaseFactory.java:50]: Very clear and easy-to-follow Clean Architecture approach. Great job!

**Review by YolandaThant — COMMENTED** [review]
Very clear and easy-to-follow Clean Architecture approach. Great job!

File: src/main/java/entity/ChatChannel.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    private String channelURL;
+    private String user1Id;
+    private String user2Id;
+    private String lastMessage;
+
+    public ChatChannel(String channelURL, String lastMessage, String user1Id, String user2Id) {
+        this.channelURL = channelURL;
+        this.lastMessage = lastMessage;
+        this.user1Id = user1Id;
+        this.user2Id = user2Id;
```
Comment (YolandaThant) [inline src/main/java/entity/ChatChannel.java:16]: Style: Perhaps we could write the getters in 1 line to mitigate overwhelming scrolls.
Double E's from me to you: Excellent Effort!

**Review by YolandaThant — COMMENTED** [review]
Style: Perhaps we could write the getters in 1 line to mitigate overwhelming scrolls.
Double E's from me to you: Excellent Effort!

File: src/main/java/interface_adapter/chat/ChatRestController.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
+package interface_adapter.chat;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
```
Comment (YolandaThant) [inline src/main/java/interface_adapter/chat/ChatRestController.java:7]: Clean and concise format. i.e. Use of final keyword where variables are not re-assigned enhances immutability & readability!

**Review by YolandaThant — COMMENTED** [review]
Clean and concise format. i.e. Use of final keyword where variables are not re-assigned enhances immutability & readability!

File: src/main/java/entity/ChatChannel.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    public String getChannelURL() {
+        return channelURL;
+    }
+
+    public String getLastMessage() {
```
Comment (YolandaThant) [inline src/main/java/entity/ChatChannel.java:23]: Style: Perhaps we could write the getters in 1 line to mitigate overwhelming scrolls.
Double E's from me to you: Excellent Effort!

**Review by YolandaThant — COMMENTED** [review]
Style: Perhaps we could write the getters in 1 line to mitigate overwhelming scrolls.
Double E's from me to you: Excellent Effort!

**Pull request merged**